plugins/remark-lint-validate-internal-links/index.js

```diff
@@ -0,0 +1,204 @@
+      log(`${ref} is handled with "no-dead-urls" from remark-lint-no-dead-urls`);
+      cache[ref] = warnings.valid;
+      return;
+    }
+    if (isLocalAnchorRef(ref)) {
+      log(`${ref} is handled with "missing-heading" from remark-validate-links`);
+      cache[ref] = warnings.valid;
+      return;
+    }
+    if (isRelativeRef(ref)) {
+      file.message(warnings.relativePath, node);
+      cache[ref] = warnings.relativePath;
+      return;
+    }
+    if (isStaticRef(ref)) {
+      // Ensure we don't allow upper case in static file names. This section
+      // does not handle images which is done in verifyImages.
+      if (hasUpperCase(ref.split('/').pop())) {
+        file.message(warnings.pathWithUpperCase, node);
+        cache[ref] = warnings.pathWithUpperCase;
+        return;
+      }
+      addWarningIfFileIsMissing({
+        file,
+        node,
+        refRoot: staticRoot,
+        ref,
+        warning: warnings.missingStaticFile
+      }) || ( cache[ref] = warnings.valid );
+      return;
+    }
+    if (isDocRef(ref)) {
+      // paths with upper cases are not working
+      if (hasUpperCase(ref)) {
+        file.message(warnings.pathWithUpperCase, node);
+        cache[ref] = warnings.pathWithUpperCase;
+        return;
+      }
+      addWarningIfFileIsMissing({
+          file,
+          node,
+          refRoot: docsRoot,
+          ref: ref.endsWith('/') ? `${ref.slice(0, -1)}.md` : `${ref}.md`,
+          originalRef: ref,
+          warning: warnings.missingDoc
+      }) || ( cache[ref] = warnings.valid );
+      return;
+    }
+    verifyAnchor(file, node, ref);
+  }
+  
+  function verifyImages(node) {
+    const ref = node.url;
+    if (hasUpperCase(ref.split('/').pop())) {
+      file.message(warnings.pathWithUpperCase, node);
+      cache[ref] = warnings.pathWithUpperCase;
+      return;
+    }
+    addWarningIfFileIsMissing({
+      file,
+      node,
+      refRoot: staticRoot,
+      ref,
+      warning: warnings.missingImage
+    }) || ( cache[ref] = warnings.valid );
+  }  
+}
+
+function verifyAnchor(file, node, ref) {
+  let [filePath, anchor] = ref.split('#');
+  if (filePath.endsWith('/')) {
+    filePath = filePath.slice(0, -1)
+  }
+  filePath = join(projectRoot, docsRoot, `${filePath}.md`);
+  const doc = getFile({ file, node, ref, filePath });
+  if (doc) {
+    let anchorFound = 0;
+    // We got the doc now turn it into an AST.
+    const tree = fromMarkdown(doc);
+    visit(tree, "heading", compare);
+    function compare(currentNode) {
+      // Here we get the current node from AST which holds the heading value but
+      // it is the original one with spaces and upper case in the beggining.
+      // The references are with lower cases only and dashes instead of spaces
+      // to be URL friendly. So we do transform what we get as heading value in
+      // order to do proper comparison.
+      const refAnchor
+        = currentNode.children[0].value.toLowerCase().replace(' ', '-');
```

Comment:
**issue**: this is barely scratching the surface of what needs to be done.  this isn't generating an anchors from a heading, it's only replacing spaces, and worse, it's only matching the first space.  this needs to do **exactly** what Hugo does when building anchors or we're going to have issues.  Anchors can clash, and when they do, they get a number increment.  For example, in the "misc.md", to target each "Duplicate (fixme)" anchor, it's "duplicate-fixme" and "duplicate-fixme-1".